src/flagsmith/mod.rs

```diff
@@ -472,4 +555,38 @@ mod tests {
         // for each subsequent refresh
         api_mock.assert_hits(3);
     }
+
+    #[test]
+    fn test_local_evaluation_identity_override_evaluate_expected() {
+        // Given
+        let environment_key = "ser.test_environment_key";
+        let response_body: serde_json::Value = serde_json::from_str(ENVIRONMENT_JSON).unwrap();
+
+        let mock_server = MockServer::start();
+        mock_server.mock(|when, then| {
+            when.method(GET)
+                .path("/api/v1/environment-document/")
+                .header("X-Environment-Key", environment_key);
+            then.status(200).json_body(response_body);
+        });
+
+        let url = mock_server.url("/api/v1/");
+
+        let flagsmith_options = FlagsmithOptions {
+            api_url: url,
+            environment_refresh_interval_mills: 100,
+            enable_local_evaluation: true,
+            ..Default::default()
+        };
+
+        // When
+        let mut _flagsmith = Flagsmith::new(environment_key.to_string(), flagsmith_options);
+        thread::sleep(std::time::Duration::from_millis(250));
```

Comment:
Is this sleep necessary to allow the local evaluation document to be ready and loaded? In other words, if we don't sleep do we have an empty document? If so, we should rework the blocking logic and block on boot for the first local evaluation like we currently do in the Python client.

Reply:
Good shout. Want to create an issue for this?

Reply:
Good catch, fixed this!